tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -2486,6 +2486,26 @@ void CGMSHLSLRuntime::AddHLSLFunctionInfo(Function *F, const FunctionDecl *FD) {
     }
   }
 
+  // Make sure that if "NodeMaxDispatchGrid" is used, that an input
+  // with the SV_DispatchGrid attribute is provided.
+  if (funcProps->Node.MaxDispatchGrid[0] > 0) {
+    bool found = false;
+    for (auto node : funcProps->InputNodes) {
+      if (node.RecordType.SV_DispatchGrid.NumComponents != 0) {
+        found = true;
+        break;
+      }
+    }
+
+    if (!found) {
+      unsigned DiagID = Diags.getCustomDiagID(
+          DiagnosticsEngine::Error,
+          "node shader '%0' with NodeMaxDispatchGrid attribute must declare an"
+          " input record containing a field with the SV_DispatchGrid semantic");
+      Diags.Report(FD->getLocation(), DiagID) << FD->getIdentifier()->getName();
```

Comment:
Where is the test for this one?

Reply:
I realize now that this code was moved in the base branch, and does have tests there now.  So why would it show up as added here?

Reply:
There were merge conflicts. I remember I had to add this code in to resolve the conflict.

Reply:
I don't think the merge conflict was resolved correctly, otherwise this shouldn't be showing up in the diff anymore.  It might be simplest at this point to rebase and squash your changes on top of the target branch, then remove this from the final squashed change if your change adds it.